packages/block-library/src/navigation-submenu/edit.js

```diff
@@ -504,6 +509,7 @@ export default function NavigationSubmenuEdit( {
 		},
 		{
 			allowedBlocks: ALLOWED_BLOCKS,
+			directInsert: DIRECT_INSERT,
```

Comment:
There are existing mechanisms in variations and blocks to define whether they can be used in the inserter. Introducing some level of granularity there might be an alternative to this prop.

e.g. 
```json
"supports": {
    "inserter": "__experimentalQuick" | "__experimentalGlobal" | true | false
}
```

Reply:
I remember @gwwar also looked into a default block for the navigation block, so going back to that might also be an option.

Reply:
The related PR was here, where we explored added a default block per container (vs one for the entire editor). https://github.com/WordPress/gutenberg/pull/29490#issuecomment-800533582

If I recall we ran into some of the same issues, where we made a guess as to what folks wanted to insert, but transforming blocks after that was more difficult.

Reply:
> Let's highlight the default block better in the quick inserter. 

🤔 From https://github.com/WordPress/gutenberg/issues/34041 it might be worth trying an `__experimentalDefaultBlock` block list setting again.

I'll try tidying that experimental branch to see if its viable.

Reply:
> There are existing mechanisms in variations and blocks to define whether they can be used in the inserter. Introducing some level of granularity there might be an alternative to this prop.

I don't think a setting in the block.json will work in this case, because we need to change it based on the block's innerBlocks. If any of the innerBlocks are _not_ Nav Links, the quick inserter should appear instead of directly inserting a Link.

For the same reason, I'm not sure we'd want to leverage @gwwar 's `__experimentalDefaultBlock`, because the Insert before and Insert after buttons would be tied to its value. If we use `__experimentalDefaultBlock`, as it stands in #29490, to tell the block inserter when we want a default block to be inserted directly, we'll end up disabling Insert before and Insert after whenever the Navigation contains anything other than Links, which is quite unexpected.

Independently of whether we do enable Insert before and Insert after on all block containers in #29490 (which I think we should! it makes total sense) we still want to make it easier to insert Nav Links inside the Nav block, and a workflow that forces us to look for a deeply hidden option in the block toolbar doesn't satisfy that requirement. 

In order to keep `useInnerBlocksProps` simple it would be nice if we could leverage the same prop for both use cases, but I'm not sure it will be possible. Am keen to know if anyone has other ideas around this though!

Reply:
That part of it is pretty complicated. There's still this part of the issue that indicates a way to designate a default block within the nav block will be needed:

> Let's highlight the default block better in the quick inserter. It should always come in first. A user should not have to think when they open a quick inserter if they are not looking for a specific block.

Maybe an option is to look at using `renderAppender` to build out a lot of the custom insertion behavior first before formalising it into props and configuration.

Reply:
I'm not sure if `__experimentalDefaultBlock` is viable, but the idea of a default block other than a global one for the editor may still be useful. (The buttons block inserter cheats a bit since only one child item is allowed and doesn't apply well to other situations). 

I mostly just rebased the branch yesterday and resolved conflicts. I'll ping y'all if I think there might be viable flow to look at.

Reply:
> Maybe an option is to look at using renderAppender to build out a lot of the custom insertion behavior first before formalising it into props and configuration.

Not sure if I'm missing something, but `renderAppender` seems to be used only for determining what the appender looks like. It doesn't feel right to mix that with its behaviour (whether it should display quick inserter or insert a block directly), because we'll want to be able to set both independently.

> There's still this part of the issue that indicates a way to designate a default block within the nav block will be needed

This is a good point. I'm thinking we could have two props: a `defaultBlock` array that specifies the default block name and any relevant attributes, and a `directInsert` boolean that specifies whether the default block, if it exists, should be inserted directly or not. That way `defaultBlock` can be used to specify the default for the quick inserter, as well as for insert before and after. If I'm not mistaken `defaultBlock` will need to be passed to `useInnerBlockProps` for all of these scenarios.

Reply:
@gwwar I'm going to steal the`__experimentalDefaultBlock` prop and give it a go in this PR 😂

Reply:
Go for it ✨